src/mlpack/tests/image_load_test.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file image_load_test.cpp
+ * @author Mehul Kumar Nirala
+ *
+ * Tests for data::LoadImage().
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <sstream>
+#include <mlpack/core.hpp>
+#include <mlpack/core/data/load_image.hpp>
+#include <boost/test/unit_test.hpp>
+
+using namespace mlpack;
+using namespace std;
+
+BOOST_AUTO_TEST_SUITE(ImageLoadTest);
+
+/**
+ * Test the invalid extension.
+ */
+BOOST_AUTO_TEST_CASE(LoadInvalidExtensionFile)
+{
+  data::LoadImage loader;
+  arma::Mat<unsigned char> img;
+  BOOST_REQUIRE_THROW(loader.Load("invalidExtendion.p4ng",
+      std::move(img)),  std::runtime_error);
+}
+
+/**
+ * Test the images is loaded correctly into the matrix.
+ */
+BOOST_AUTO_TEST_CASE(LoadImageIntoMatrixFromFile)
+{
+  data::LoadImage loader;
+
+  // Matrix to load contents of file into.
+  arma::Mat<unsigned char> img;
+  BOOST_REQUIRE(loader.Load("test_image.png", std::move(img)) == true);
```

Comment:
I can't comment on the image file itself, but to keep the repo size low, do you think we could resize the image to reduce the size?

Reply:
I guess it would be better to add in the cmake. Will that be okay?

Reply:
The image file? I think it would be fine if we resize the image locally and push the updated version, I guess we could also tar the file as well, not sure that is needed. Right now the file is 463 KB, which I think could be reduced to 50 KB or something in that region, since we don't really care how the image looks like?